structuredheadings/plugin.js

```diff
@@ -326,19 +326,30 @@
             func = this.clearAutonumberClassesForHeading;
           } else {
             func = this.setAutonumberClassesForHeading.bind(this, value);
+
+             // prep the html for the collapsed, not in a heading case
+            if (headings === null) {
+              // avoid an extra snapshot created by the matchHeading command
+              editor.fire("lockSnapshot", { dontUpdate: true });
+              editor.execCommand("matchHeading");
+              editor.fire("unlockSnapshot");
+              // put the new heading into the headings array
+              headings = [CKEDITOR.plugins.structuredheadings.getCurrentBlockFromPath(editor)];
+            }
```

Comment:
this method in general is getting a bit long, but I'm going to take a "we'll do a big refactor if we need to" approach.  Probably, more things need to be encapsulated as commands, (perhaps with some more duplication code) and this click handler should simply detect which command to fire based on the selection type